### PR TITLE
SONARJAVA-1619 FP on anonymous classes not redefining some methods

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/TooManyMethodsCheck.java [hunk 1]
```diff
@@ -23,15 +23,19 @@
...
+import org.sonar.java.resolve.JavaSymbol;
 import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
 import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.semantic.Symbol;
 import org.sonar.plugins.java.api.tree.ClassTree;
 import org.sonar.plugins.java.api.tree.MethodTree;
 import org.sonar.plugins.java.api.tree.NewClassTree;
 import org.sonar.plugins.java.api.tree.Tree;
 
 import java.util.ArrayList;
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/TooManyMethodsCheck.java]: MINOR Remove this unused import 'java.util.ArrayList'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: java-checks/src/main/java/org/sonar/java/checks/TooManyMethodsCheck.java [hunk 2]
```diff
@@ -23,15 +23,19 @@
 import org.sonar.check.Rule;
 import org.sonar.check.RuleProperty;
 import org.sonar.java.checks.helpers.ExpressionsHelper;
+import org.sonar.java.model.expression.MethodInvocationTreeImpl;
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/TooManyMethodsCheck.java]: MINOR Remove this unused import 'org.sonar.java.model.expression.MethodInvocationTreeImpl'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: java-checks/src/main/java/org/sonar/java/checks/TooManyMethodsCheck.java [hunk 3]
```diff
@@ -87,4 +89,9 @@ public void visitNode(Tree tree) {
     }
   }
 
+  private boolean isOverriding(MethodTree member) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/TooManyMethodsCheck.java]: MAJOR Make "isOverriding" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: java-checks/src/main/java/org/sonar/java/checks/TooManyMethodsCheck.java [hunk 4]
```diff
@@ -59,32 +61,44 @@
...
+      secondary,
+      null);
+  }
+
+  private boolean shouldNotReportIssue(ClassTree classTree, List<Tree> count) {
+    return (classTree.simpleName() == null && count.stream().filter(member -> !isOverriding((MethodTree) member)).count() == 0)
+      ||  count.size() <= maximumMethodThreshold;
+  }
+
+  private static boolean isOverriding(MethodTree member) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/TooManyMethodsCheck.java]: MAJOR Make "isOverriding" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: java-checks/src/main/java/org/sonar/java/checks/TooManyMethodsCheck.java [hunk 5]
```diff
@@ -59,32 +61,44 @@
...
     }
+    reportIssue(
+      ExpressionsHelper.reportOnClassTree(classTree),
+      String.format("%s has %d%s methods, which is greater than the %d authorized. Split it into smaller classes.",
+        classDescription, methods.size(), countNonPublic ? "" : " public", maximumMethodThreshold),
+      secondary,
+      null);
+  }
+
+  private boolean shouldNotReportIssue(ClassTree classTree, List<Tree> count) {
```
Comment (tvsonar) [inline java-checks/src/main/java/org/sonar/java/checks/TooManyMethodsCheck.java]: `count` should be named `methods`

Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/TooManyMethodsCheck.java]: indeed. Hit and missed refactoring.

File: java-checks/src/test/java/org/sonar/java/checks/TooManyMethodsCheckTest.java [hunk 6]
```diff
@@ -22,13 +22,14 @@
 import org.junit.Test;
 import org.sonar.java.checks.verifier.JavaCheckVerifier;
 
-public class TooManyMethodsCheckTest {
+public class  TooManyMethodsCheckTest {
```
Comment (tvsonar) [inline java-checks/src/test/java/org/sonar/java/checks/TooManyMethodsCheckTest.java]: There's an unnecessary additional space here

**Pull request merged**